runner_service/inventory.py

```diff
@@ -91,37 +99,52 @@ def load(self):
 
         if not os.path.exists(self.filename):
             try:
-                with open(self.filename, 'w') as inv:
+                # Using Python 3 exclusive creation
+                with open(self.filename, 'x') as inv:                    
                     inv.write(yaml.dump(AnsibleInventory.inventory_seed,
-                                        default_flow_style=False))
+                                            default_flow_style=False))
+            except FileExistsError:
+                logger.info("Inventory file '{}' already created".format(self.filename))
             except IOError:
                 raise InventoryWriteError("Unable to create the seed inventory"
                                           " file at {}".format(self.filename))
-
-        if self.exclusive_lock:
-
-            try:
-                self.fd = open(self.filename, 'r+')
-                self.lock()
-            except (BlockingIOError, OSError):
-                # Can't obtain an exclusive_lock
-                self.fd.close()
-                return
+            
+        try:    
+            if self.exclusive_lock:
+                try:
+                    self.fd = open(self.filename, 'r+')
+                    self.lock()
+                except (BlockingIOError, OSError):
+                    # Can't obtain an exclusive_lock
+                    self.fd.close()
+                    return
+                else:
+                    raw = self.fd.read().strip()
             else:
-                raw = self.fd.read().strip()
-        else:
-            raw = fread(self.filename)
+                raw = fread(self.filename)
+        except Exception as ex:
+            raise InventoryreadError("Unable to read the inventory"
+                                     " file at {}, error: {}".format(self.filename, ex))
 
         if not raw:
-            self.inventory = None
+            # If the inventory is empty for some extrange reason
+            self.inventory = AnsibleInventory.inventory_seed
```

Comment:
I don't think we should do this. If we can't read the inventory - we can't just seed it - we need to ensure the file is consistent with what we're working against. If we can't read it - we flag it and abort. What if the inventory did contain 'stuff' but the 'raw' was empty - we'd lose what was there!

Reply:
I guess given the logic I'm saying that I can't think of a scenario where raw would be empty - if it is I'd rather return None, so the inventory load will end up throwing an error to the caller.

Reply:
Ok. i'll change :-)